2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -177,21 +177,92 @@ function run_mongod_supervisor() {
   ${CONTAINER_SCRIPTS_PATH}/replica_supervisor.sh 2>&1 &
 }
 
-# mongo_create_users creates the MongoDB admin user and the database user
-# configured by MONGODB_USER
-function mongo_create_users() {
-  mongo ${MONGODB_DATABASE} --eval "db.addUser({user: '${MONGODB_USER}', pwd: '${MONGODB_PASSWORD}', roles: [ 'readWrite' ]})"
+# mongo_create_admin creates the MongoDB admin user with password: MONGODB_ADMIN_PASSWORD
+# $1 - login parameters for mongo (optional)
+# $2 - host where to connect (localhost by default)
+function mongo_create_admin() {
+  if [[ -z "${MONGODB_ADMIN_PASSWORD:-}" ]]; then
+    echo "=> MONGODB_ADMIN_PASSWORD is not set. Authentication can not be set up."
+    exit 1
+  fi
+
+  # Set admin password
+  set +e
+  mongo admin ${1:-} --host ${2:-"localhost"} --eval "db.addUser({user: 'admin', pwd: '${MONGODB_ADMIN_PASSWORD}', roles: ['dbAdminAnyDatabase', 'userAdminAnyDatabase' , 'readWriteAnyDatabase','clusterAdmin' ]});"
```

Comment:
@omron93 if you move this line to the if condition [a few lines below](https://github.com/openshift/mongodb/pull/136/files#diff-1763ec43870f7abf682b1057cd2daa84R195), then you don't need to `set +e; set -e`, nor the `result` variable.

Like this:

```bash
if ! mongo admin ...; then
  echo "=> Failed to create MongoDB admin user."
  exit 1
fi
```

Since the line is pretty long, you might want to store the string we pass to `--eval` as a variable, just like you did [here](https://github.com/openshift/mongodb/pull/135/files#diff-1763ec43870f7abf682b1057cd2daa84R62). Up to you :)